### PR TITLE
Add pid_of() to get the pid of a process by name

### PR DESCRIPTION
**Pull request by abhiarora4**
* Add pid_of() function to get the pid of a process having "exe_name" passed to it as parameter.
* Really handy call. Alternate to calling `system("pidof abcd")`.

### CODE REVIEW

File: src/procutils.c [hunk 1]
```diff
@@ -89,3 +94,43 @@ int o_redirect(int mode, const char *file)
...
+        return 0;
+    struct dirent *ent;
+    while ((ent = readdir(dir)) != NULL)
+    {
+        int pid = 0;
+        const char *ptr = ent->d_name;
+        if (ptr == NULL)
+            continue;
+
+        for (; *ptr && isdigit(*ptr); ptr++) // Convert string to integer
```
Comment (sidcha) [inline src/procutils.c:111]: Perhaps you should user strtoul()?

File: src/procutils.c [hunk 2]
```diff
@@ -89,3 +94,43 @@ int o_redirect(int mode, const char *file)
...
+            pid = pid * 10 + (*ptr - '0');
+        if (*ptr != '\0')
+            continue;
+
+        char cmdline_file[(ptr - ent->d_name) + 16];
```
Comment (sidcha) [inline src/procutils.c:116]: Can you host this and define it to PATH_MAX?

File: src/procutils.c [hunk 3]
```diff
@@ -89,3 +94,43 @@ int o_redirect(int mode, const char *file)
...
+        snprintf(cmdline_file, sizeof(cmdline_file), "/proc/%s/cmdline", ent->d_name);
+
+        FILE *file = fopen(cmdline_file, "r");
+        if (file == NULL)
+            continue;
+
+        char name[200] = { 0 };
```
Comment (sidcha) [inline src/procutils.c:123]: You can reuse the same buffer for here too. call it some thing generic like "buf".

File: src/procutils.c [hunk 4]
```diff
@@ -89,3 +94,43 @@ int o_redirect(int mode, const char *file)
...
+        bool converted = fscanf(file, "%199s", name) == 1;
```
Comment (sidcha) [inline src/procutils.c:124]: Why not strncpy()? fscanf() for getting the entire string seems excessive.

File: tests/test-procutils.c [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+
+int test_pid_of(void)
+{
+    FILE *pfile = fopen("/proc/self/status", "r");
+    if (pfile == NULL)
+        return -1;
+
+    char filename[200] = { 0 };
+    fscanf(pfile, "Name: %199s", filename);
```
Comment (sidcha) [inline tests/test-procutils.c:19]: For instance, fscanf() here makes sense. :)

File: tests/test-procutils.c [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    fclose(pfile);
+
+    int pid = pid_of(filename);
+    return pid > 0 ? 0 : -1;
```
Comment (sidcha) [inline tests/test-procutils.c:23]: Check it agains getpid()?

**Review by sidcha — COMMENTED** [review]
Looks good. But,

- overall has some indentation issues. Use tabs.
- commit message does not need to have the [FEATURE ADDITION] 
- commit message needs to be imperative tone. eg: "Add pod_of() to get pid of a process by name"

File: src/procutils.c (hunk 4 above)
Comment (abhiarora4) [reply src/procutils.c:124]: `fscanf` is reading the line from the file. We can't replace it with `strncpy`?

**Review by sidcha — APPROVED** [review]
LGTM, thanks!

**Pull request merged**